packages/contracts/test/exchange/internal.ts

```diff
@@ -0,0 +1,329 @@
+import { BlockchainLifecycle } from '@0xproject/dev-utils';
+import { Order, SignedOrder } from '@0xproject/types';
+import { BigNumber } from '@0xproject/utils';
+import * as _ from 'lodash';
+
+import { TestExchangeInternalsContract } from '../../generated_contract_wrappers/test_exchange_internals';
+import { artifacts } from '../utils/artifacts';
+import { getGanacheOrGethError } from '../utils/assertions';
+import { chaiSetup } from '../utils/chai_setup';
+import { bytes32Values, testCombinatoriallyWithReferenceFuncAsync, uint256Values } from '../utils/combinatorial_utils';
+import { constants } from '../utils/constants';
+import { provider, txDefaults, web3Wrapper } from '../utils/web3_wrapper';
+
+chaiSetup.configure();
+const blockchainLifecycle = new BlockchainLifecycle(web3Wrapper);
+
+const MAX_UINT256 = new BigNumber(2).pow(256).minus(1);
+
+const emptyOrder: Order = {
+    senderAddress: '0x0000000000000000000000000000000000000000',
+    makerAddress: '0x0000000000000000000000000000000000000000',
+    takerAddress: '0x0000000000000000000000000000000000000000',
+    makerFee: new BigNumber(0),
+    takerFee: new BigNumber(0),
+    makerAssetAmount: new BigNumber(0),
+    takerAssetAmount: new BigNumber(0),
+    makerAssetData: '0x',
+    takerAssetData: '0x',
+    salt: new BigNumber(0),
+    exchangeAddress: '0x0000000000000000000000000000000000000000',
+    feeRecipientAddress: '0x0000000000000000000000000000000000000000',
+    expirationTimeSeconds: new BigNumber(0),
+};
+
+const emptySignedOrder: SignedOrder = {
+    ...emptyOrder,
+    signature: '',
+};
+
+interface FillResults {
```

Comment:
Can we add this to `types.ts`? I'm guessing we'll use it elsewhere at some point.

Reply:
👍 It was actually already there so I just imported it.